tests/posix/common/src/mutex.c

```diff
@@ -195,4 +195,14 @@ ZTEST(mutex, test_mutex_timedlock)
 	zassert_ok(pthread_mutex_destroy(&mutex));
 }
 
-ZTEST_SUITE(mutex, NULL, NULL, NULL, NULL, NULL);
+static void before(void *arg)
```

Comment:
This pattern seems a little boilerplatey.  Maybe a nonstatic "test_check_dyn_thread()" or something instead?

Reply:
As much as I would like to, I just want to help this PR crawl past the finish line 😅 - another PR maybe?